include/seastar/util/string_utils.hh

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2015 Cloudius Systems
+ */
+
+//
+// request.hpp
+// ~~~~~~~~~~~
+//
+// Copyright (c) 2003-2013 Christopher M. Kohlhoff (chris at kohlhoff dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#ifndef SEASTAR_MODULE
+#include <cstring>
+#include <stdio.h>
+#endif
+
+#include <seastar/util/modules.hh>
+#include <seastar/core/sstring.hh>
+
+namespace seastar {
```

Comment:
Plz, also wrap it with `namespace internal`. Or put the whole thing into http/common.{cc|hh}

Reply:
Interesting, why do you think a utility like this should be marked `internal`? any implementation detail of seastar to hide here?
Also adding some guidance on that in `contributing.md` would be great I guess.

Reply:
No details to hide. If we add anything to non-internal API it should be
- documented
- maintained in backward compatible manner
The former is not implemented here, the latter is something unlikely to cause any troubles, but still